### ISSUE TITLE
SIM115 open() inside a return statement not reported

### ISSUE THREAD

**Issue by madp** [astral-sh/ruff#25916]
### Summary

SIM115 fails to detect open() when it appears inside a return statement:

```
def not_detected():
    return [x for x in open("a")]
```

```
def detected():
    result = [x for x in open("a")]
```

### Version

ruff 0.15.17

**Comment by ntBre** [comment]
I think that's an intentional exclusion to the rule:

https://github.com/astral-sh/ruff/blob/a4f73ff530306cee4734038469333f9d8b1f5bc0/crates/ruff_linter/src/rules/flake8_simplify/rules/open_file_with_context_handler.rs#L245-L248

because returning from a context manager would close the file. The rule assumes that you are intentionally returning the file handles to the caller.
